### PR TITLE
Feature/kinetics

### PR DESCRIPTION
**Pull request by atravert**
improvement of integration speed

**Checklist**:

- [x] Close the #647
- [X] Tests have been adapted

### CODE REVIEW

File: spectrochempy/analysis/kinetic_utilities.py [hunk 1]
```diff
@@ -44,14 +61,15 @@ class SolverError(SpectroChemPyError):
...
+    # transform an equation given as a string to dictionaries of species with
+    # integer stoechiometric coefficients for the left (reactants) and right (products)
+    # side of the equation.
 
     regex = r"(((([\.,0-9]*)((?=[a-zA-Z])[a-zA-Z, 1-9]+))(?=\+?))?(?=(->|→)?))"
 
     matches = re.finditer(regex, eq.replace(" ", ""))
-    equation = {}
-    mult = -1  # reactants
+    left, right = {}, {}
```
Comment (atravert) [inline spectrochempy/analysis/kinetic_utilities.py:71]: A and B matrices (left and right stoichiometries) must be determined separately, else reactions such as A + B = 2 A + C are not handled properly

File: spectrochempy/analysis/kinetic_utilities.py [hunk 2]
```diff
@@ -63,18 +81,22 @@ def _interpret_equation(eq, species):
                 f"Available species : {species}"
             )
         coef = match.group(4) if match.group(4) else 1
-        equation[s] = float(coef) * mult
+        try:
+            coef = int(coef)
```
Comment (atravert) [inline spectrochempy/analysis/kinetic_utilities.py:85]: As we deal with elementary steps, stoichiomtruc coefficicients must be int

File: spectrochempy/analysis/kinetic_utilities.py [hunk 3]
```diff
@@ -63,18 +81,22 @@ def _interpret_equation(eq, species):
...
+
         if match.group(6) in ["->", "→"]:
             # shift to products
-            mult = 1
-
-    # normalize the coefficients (for float we assume that they have at the maximum
-    # 2 decimals
-    y = (np.array(list(equation.values())) * 100).astype(int)
-    div = np.gcd.reduce(np.abs(y))
-    equation = {k: int(v * 100 / div) for k, v in equation.items()}
```
Comment (atravert) [inline spectrochempy/analysis/kinetic_utilities.py:75]: normalization is not expected as we are dealing with elementary steps

File: spectrochempy/analysis/kinetic_utilities.py [hunk 4]
```diff
@@ -108,33 +130,35 @@ class ActionMassKinetics(tr.HasTraits):
...
         # initialise concentrations, species and equations
         self._concentrations = species
         self._species = list(self._concentrations.keys())
         self._equations = equations
+        self._production_rates = self._write_production_rates()
 
     # ----------------------------------------------------------------------------------
     # Private methods
     # ----------------------------------------------------------------------------------
     @tr.validate("_k")
```
Comment (fernandezc) [inline spectrochempy/analysis/kinetic_utilities.py:152]: The most important here for the validation to occur is to change the string in @tr.validate("_k"). The following func name has no importance except for clarity

File: spectrochempy/analysis/kinetic_utilities.py (hunk 3 above)
Comment (fernandezc) [reply spectrochempy/analysis/kinetic_utilities.py:75]: Ok I understand, but what about using a scaling factor to transform into int. so the eq. could be write either way (int or not)?  but ok no problem with this.

Comment (atravert) [reply spectrochempy/analysis/kinetic_utilities.py:75]: Actually,  the rate expression for 2 A -> 2B  (r = k C_A**2) differs from that of A -> B  (r = k C_A), so it is safer to not normalize)

File: spectrochempy/analysis/kinetic_utilities.py (hunk 4 above)
Comment (atravert) [reply spectrochempy/analysis/kinetic_utilities.py:152]: my mistake !

**Comment by fernandezc** [comment]
Should we merge or are still working on this PR?

**Comment by fernandezc** [comment]
> Should we merge or are you still working on this PR?

@atravert , If you agree to merge and include this in the next release 0.6.4, could you add it to changelog ? 
Thanks

**Comment by fernandezc** [comment]
Test on windows does not pass anymore, because the merge with master. see the problem described here https://github.com/spectrochempy/spectrochempy/pull/650#issuecomment-1551283267. 

If this do not work I may just stop trying to check the scripts on windows.

**Comment by fernandezc** [comment]
> > Should we merge or are you still working on this PR?
> 
> @atravert , If you agree to merge and include this in the next release 0.6.4, could you add it to changelog ? Thanks

Thanks for merging this. Should I make a new release as we have significative changes (LinearCoord, Kinetics)?

**Pull request merged**